fuzz_lightyear/supplements/factory.py

```diff
@@ -70,16 +88,36 @@ def register_factory(keys: Union[str, Iterable[str]]) -> Callable:
     """
     # This is renamed just to make mypy happy.
     _keys = listify_decorator_args(keys)
+    _endpoints = listify_decorator_args(endpoints)
 
     def decorator(func: Callable) -> Callable:
         wrapped = inject_user_defined_variables(func)
 
         mapping = get_user_defined_mapping()
         for key in _keys:
             if key in mapping:
-                raise ConflictingKeys(key)
-
-            mapping[key] = wrapped
+                func_dict = mapping[key]
+                if _endpoints:
+                    for endpoint in _endpoints:
+                        if endpoint in func_dict:
+                            raise ConflictingKeys(key)  # TODO: raise different error?
```

Comment:
Might want to modify the `ConflictingKeys` exception to take in an optional `endpoint` argument, so we can show a better error message.